PeriodogramAnalysis/test_spectrogram_analysis.py

```diff
@@ -52,22 +40,36 @@ def test_call_return_shape():
     assert sxx_list.shape == (2, frequencies.shape[0], times.shape[0])
 
 
-def test_computing_spectrum() -> None:
+def test_computing_spectrum(mock_signal_generator, mocker) -> None:
     power_analysis = SpectrogramAnalysis()
-    signal_gen = signal_input()
-    frequencies, times, sxx_list = power_analysis(signal_gen)
-
-    for ch in range(signal_gen.data.shape[1]):
-        channel_data = signal_gen.data[:, ch]
-
-        signal_no_bias = channel_data - np.mean(channel_data)
-        freqs_expected, times_expected, sxx_expected = spectrogram(
-            signal_no_bias,
-            fs=signal_gen.rate,
-            nperseg=int(signal_gen.rate * power_analysis.nperseg_ratio),
-            noverlap=int((int(signal_gen.rate * power_analysis.nperseg_ratio)) / 2),
+    signal, _ = mock_signal_generator
+
+    mock_spectrogram = mocker.patch("scipy.signal.spectrogram")
+
+    mock_spectrogram.return_value = (
+        np.random.rand(13),
```

Comment:
what is this? Avoid using random in any unittest. Also what is 13 and 76?

Reply:
Added explanation of 13 and 76, and I used np.ones now, it should still be representative. They are in commit [9e1fd9b]